File: src/main/java/com/OneOfManySimons/map/Collisions.java
package com.OneOfManySimons.map;


import com.OneOfManySimons.Debug;
import com.OneOfManySimons.graphics.Icons;

import javax.swing.*;

/**
 * Collision related.<br>
 * Checking, if that tile have collisions or not.
 */
public class Collisions {
	// Collision types
	public final int empty = 0;
	public final int immovable = 1;

	/**
	 * Checking, if tile has a collision border.
	 *
	 * @param tile that we want to check
	 * @return integer
	 */
	public int checkForCollision(ImageIcon tile) {
		if (Debug.map.Collisions) System.out.println("--- [Collisions.checkForCollision]");

		if (tile.equals(Icons.Environment.wall)) {
			return immovable;
		}
		if (tile.equals(Icons.Environment.blank)) {
			return immovable;
		}
		return empty;
	}
}

File: composeApp/src/commonMain/kotlin/com/one_of_many_simons/one_button_game/map/Collisions.kt
package com.one_of_many_simons.one_button_game.map

import androidx.compose.ui.graphics.ImageBitmap
import com.one_of_many_simons.one_button_game.Debug
import com.one_of_many_simons.one_button_game.graphics.Icons

/**
 * Collision related.<br></br>
 * Checking, if that tile have collisions or not.
 */
class Collisions {
    // Collision types
    val empty: Int = 0
    val immovable: Int = 1

    private val immovableTiles: Array<ImageBitmap> = arrayOf(
        Icons.Environment.wall,
        Icons.Environment.blank
    )

    /**
     * Checking, if tile has a collision border.
     *
     * @param tile that we want to check
     * @return integer
     */
    fun checkForCollision(tile: ImageBitmap): Int {
        if (Debug.Map.COLLISIONS) println("--- [Collisions.checkForCollision]")

        if (immovableTiles.contains(tile)) {
            return immovable
        }

        return empty
    }
}
